src/EnergyPlus/ZoneContaminantPredictorCorrector.cc

```diff
@@ -2608,6 +2618,157 @@ namespace ZoneContaminantPredictorCorrector {
 
                 ZoneAirCO2(ZoneNum) = ZoneAirCO2Temp(ZoneNum);
 
+				// Hybrid modeling with CO2 concentration starts here, added by Han Li 2018
+				GetHybridModelZone();
+				if (HybridModelZone(ZoneNum).InfiltrationCalc_C || HybridModelZone(ZoneNum).PeopelCountCalc_C
+					&& (!WarmupFlag) && (!DoingSizing)) {
+
+					Zone(ZoneNum).ZoneMeasuredCO2Concentration = GetCurrentScheduleValue(HybridModelZone(ZoneNum).ZoneMeasuredCO2ConcentrationSchedulePtr);
+
+					if (DayOfYear >= HybridModelZone(ZoneNum).HybridStartDayOfYear && DayOfYear <= HybridModelZone(ZoneNum).HybridEndDayOfYear) {
+
+						ZoneAirCO2(ZoneNum) = Zone(ZoneNum).ZoneMeasuredCO2Concentration;
+
+						if (HybridModelZone(ZoneNum).InfiltrationCalc_C && UseZoneTimeStepHistory) {
+							Real64 AA(0.0); // Sum of air mass flow terms except infiltration. (AA = A - M_inf)
```

Comment:
Use SumOfAirMassExpInfil instead of AA?

Reply:
The AA, BB, CC, DD are analogs to A, B, C, D in the 3rd backward differential equations in the normal simulation. Do we want to change them?